src/RIGID/fix_rigid_small.cpp

```diff
@@ -1520,8 +1547,12 @@ void FixRigidSmall::create_bodies()
     if (bbox[i][0] == bbox[i][1] && bbox[i][2] == bbox[i][3] &&
         bbox[i][4] == bbox[i][5]) flag = 1;
   MPI_Allreduce(&flag,&flagall,1,MPI_INT,MPI_SUM,world);
-  if (flagall)
-    error->all(FLERR,"One or more rigid bodies are a single particle");
+  if (flagall) {
```

Comment:
I don't understand why you are differentiating between selecting by molecule id or a custom integer parameter. The molecule id is effectively the same. That said, accepting rigid bodies with a single particle is problematic unless the particle is an extended particle. How can we have rotation for a point particle? Allowing this would also require treating this special case when computing the degrees of freedom.

Reply:
Atoms in a molecule can have different values of the custom parameter. This facilitates the creation of molecules with internal parts (e.g. rings) that behave as rigid bodies. Accepting "single-particle bodies" would also facilitate having rigid and flexible parts in the same molecule. I used quotation marks because they are, in fact, free atoms. The fix rigid integrators are already able to deal with them due to the way `MathExtra::angmom_to_omega` is written. In this case, all moments of inertia will be zero and there will be no rotation at all. Of course, this is not computationally efficient and perhaps it is a better idea to separate the atoms into two groups, rigid and flexible, and apply different fixes for them. A discount in degrees of freedom is still needed, in the same way it is done for linear bodies.